server.js

```
@@ -16,9 +25,104 @@ const app = express();
+        };
+      }
+      return endpoint;
+    });
+    res.status(200).json(endpoints);
+  } catch (error) {
+    console.error("The following error occured:", error);
+    res
+      .status(500)
+      .send(
+        "Sorry, this page is not available at the moment. Please try again later."
+      );
+  }
+});
+
+//GET the thoughts endpoint
+//Also sorting them in a decending order and limit the thoughts to 20 shown
+app.get("/thoughts", async (req, res) => {
+  try {
+    const allThoughts = await Thought.find()
+      .sort({ createdAt: -1 })
+      .limit(20)
+      .exec();
+
+    if (allThoughts.length > 0) {
+      res.status(200).json(allThoughts);
+    } else {
+      res.status(404).send("No Thoughts were found.");
+    }
+  } catch (error) {
+    console.error("The following error occured:", error);
+    res
+      .status(500)
+      .send(
+        "Sorry, this page is not available at the moment. Please try again later."
+      );
+  }
+});
+
+//POST the thoughts endpoint
+app.post("/thoughts", async (req, res) => {
+  const { message, hearts, createdAt, _id } = req.body;
+
+  try {
+    const thought = await new Thought({
+      message,
+      hearts,
+      createdAt,
+      _id,
+    }).save();
+    res.status(201).json(thought);
+  } catch (error) {
+    res
+      .status(400)
+      .send(
+        "There has been an error with posting your thought. Please ensure that your message is at least 5 characters long and has a maximum of 140 characters."
+      );
+  }
+});
+
+//POST Endpoint for adding a heart
+app.post("/thoughts/:thoughtId/like", async (req, res) => {
+  const { thoughtId } = req.params;
+
+  try {
+    const likeThought = await Thought.findByIdAndUpdate(
+      thoughtId,
+      { $inc: { hearts: 1 } },
+      { new: true, runValidators: true }
+    );
+
+    if (likeThought) {
+      res.status(200).json(likeThought);
+    } else {
+      res
```

I understand that you want to customise error handling when id is not found, which is great, but the fact is the 404 line of code will never run because if no post is found with a given id, the findByIdAndUpdate will immediately throw an error instead of returning null/undefined falsy value. Then when you input invalid id it will always show 500 error you defined in the catch block.